internal/bminventory/inventory.go

```diff
@@ -5551,3 +5553,39 @@ func (b *bareMetalInventory) V2PostStepReply(ctx context.Context, v2Params insta
 func (b *bareMetalInventory) V2GetHost(ctx context.Context, params installer.V2GetHostParams) middleware.Responder {
 	return installer.NewV2GetHostNotImplemented()
 }
+
+func (b *bareMetalInventory) BindHost(ctx context.Context, params installer.BindHostParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	log.Infof("Bind host %s to cluster %s", params.HostID, params.BindHostParams.ClusterID)
+	host, err := common.GetHostFromDB(b.db, params.InfraEnvID.String(), params.HostID.String())
+	if err != nil {
+		log.WithError(err).Errorf("failed to find host <%s> in infraEnv <%s>",
+			params.HostID, params.InfraEnvID)
+		return common.NewApiError(http.StatusNotFound, err)
+	}
+	if host.ClusterID != nil {
```

Comment:
Is this even possible? a host with InfraEnvId and ClusterId?

Reply:
If it's possible, consider adding: 
```suggestion
	if host.ClusterID != nil {
	    if host.ClusterID == *params.BindHostParams.ClusterID {
	        return 204
	    }
```

Reply:
It is possible to have both, InfraEnvID always exists since it's part of the identifier of the host (PK), if the host is currently part of a cluster then ClusterID will also have values.
Since this is an imperative action, I thought that it will be right to return an error if it's already bound and the user asks to bind even if the request is to bind to the cluster it's currently bounded to.
Anyhow, we defined that this API return Host object then I don't see a reason in some cases to return 204

Reply:
1. @eranco74 , I think you may have gotten confused by your PR, where the unit tests checked if infraenv was nil.  That's just confusing test code and there is a task to fix it.  A host always belongs to the InfraEnv whose ISO it booted from.  It can also optionally be bound to a cluster.
2. I think the original code was good as is.